### DISCUSSION TITLE
Condition для хранения Нозологической единицы Диагноза

### DISCUSSION THREAD

**Discussion by OlegPenzin** [fhir-ru/core#72] [Клиническая РГ] [answered]
**Нозологическая единица (форма)** - совокупность клинических, лабораторных и инструментальных диагностических признаков, позволяющих идентифицировать заболевание (отравление, травму, физиологическое состояние) и отнести его к группе состояний с общей этиологией и патогенезом, клиническими проявлениями, общими подходами к лечению и коррекции состояния.

Если рассматриваем внутри клинически диагнозов, то используем стандартную категорию EncounterDiagnosis
![image](https://user-images.githubusercontent.com/60599571/149512232-0ebd7583-255e-45ad-bef0-828eba10c715.png)

* Нозологическая единица
* Статус нозологической единицы в соответствии с механизмом переходов статусов (активная нозологическая единица: установленная впервые в жизни, установленная повторно: рекуррентное заболевание, рецидив заболевания; неактивная нозологическая единица: снятая(отменённая), перенесённое заболевание, ремиссия заболевания)
	** Заболевание установлено впервые в жизни или повторно (рецидивы и рекуррентные заболевания).
	* Характер заболевания (острое, хроническое).
	* Статус подтверждения нозологической единицы: подозреваемое (предварительное), подтвержденное, опровергнутое.
	* Дата установления нозологической единицы (предлагается системой, возможно изменение во временных рамках, определяемых текущим функциональным профилем модуля).
	* Код (шифр) заболевания по справочнику МКБ-10.
	* Уточнение нозологических форм по синонимам и/или клиническим расширениям МКБ-10 (ICD10-CM)
	* Локализация заболевания
	* Взаимоотношение добавляемой нозологической единицы с другими, уже присутствующими в диагнозе
	* Клиническая формулировка, выбранная лечащим врачом 
	* Указание гистологического типа новообразования, подтвержденного патологоанатомом, и его шифр по справочнику МКБ-О
	* Указание актуального состояния опухолевого процесса по системе TNM
		* режим стадирования (c yc p yp r) 
		* T (tumor) — распространённость первичной опухоли
		* N (nodus) — наличие метастазов в регионарных лимфатических узлах
		* M (metastasis) — наличие отдалённых метастазов
		* дополнительные классификации GLVR
	* Дополнительные классификаторы и параметры
	* Статус нозологической единицы в соответствии с механизмом переходов статусов (активная нозологическая единица: установленная впервые в жизни, установленная повторно: рекуррентное заболевание, рецидив заболевания; неактивная нозологическая единица: снятая(отменённая), перенесённое заболевание, ремиссия заболевания)

**Comment by eugenekogan-spb** [comment]
А что такое 

> Взаимоотношение добавляемой нозологической единицы с другими, уже присутствующими в диагнозе

?
Тут есть несколько вариантов:
- уточнение
--- данный диагноз 2 уточняет ранее выставленный диагноз 1 (то есть, и раньше у пациента было заболевание 2, но мы это не знали или ошибались)
--- данный диагноз 2 является новым состоянием той же болезни пациента 1, просто она изменилась со временем
- причина-следствие
---диагноз 2 является причиной ранее установленного 1
---диагноз 2 является осложнением ранее установленного 1

**Comment by Vrodvi** [comment]
Материалы для оперативного обсуждения структуры нозологической единицы предлагаем готовить в гугл-таблицах:
- https://docs.google.com/spreadsheets/d/1WDC22c-qnWYEJZmdKcfuIqz_OvZaC_z4b5Vg5R5YSnI/edit?usp=sharing

В таком будет возможность оставить прицельные комментарии по элементам структуры, а также возможность обсудить и скорректировать структуру во время проведения рабочей группы.

**Reply by eugenekogan-spb** [reply]
Дайте право редактирования, напишу комментарии в правой колоночке

**Reply by Vrodvi** [reply]
Добавил право на редактирование.

**Reply by Vernir** [reply]
Ссылка на доску в миро по статусам нозологической единицы

https://miro.com/app/board/o9J_lOSevQE=/?invite_link_id=654270459135

**Comment by OlegPenzin** [comment]
Примеры связей между Condition и Observation из опыта коллег:

- https://hl7.org/fhir/us/mcode/Condition-primary-cancer-condition-breast.html
-- https://hl7.org/fhir/us/mcode/Observation-tnm-clinical-stage-group-3c.html
--- https://hl7.org/fhir/us/mcode/Observation/tnm-clinical-primary-tumor-category-cT3
--- https://hl7.org/fhir/us/mcode/Observation/tnm-clinical-regional-nodes-category-cN3
--- https://hl7.org/fhir/us/mcode/Observation/tnm-clinical-distant-metastases-category-cM0

**Reply by eugenekogan-spb** [reply]
Спасибо, пример доказывает, что коллеги так делают. 
Однако, моя мысль была в следующем. Атрибут Condition.Stage показывает стадию патологического (физиологического) процесса. В поле summary записано значение стадии (как это сделано и в mcode). В поле assessment - Reference to a formal record of the evidence on which the staging assessment is based.
Подтверждением высказывания о стадии могут быть сведения об осмотре врача, выявленные симптомы, данные обследования. Но подтверждением высказывания о стадии не может быть само высказывание о стадии.
То есть, ссылаясь на Observation, в котором в Value записано то же значение стадии, мы ничего не подтверждаем. Это аналогично тому, что мы просто не заполнили поле Assessment.

**Reply by Vrodvi** [reply]
Подтверждением стадии онкологического процесса являются оценки по следующим характеристикам:
T (tumor) — распространённость первичной опухоли
N (nodus) — наличие метастазов в регионарных лимфатических узлах
M (metastasis) — наличие отдалённых метастазов

Данные характеристики "формализовано записаны" (formal record) в виде трёх ресурсов Observation, дополнительно создана "формализованная запись" о стадии на основании этих характеристик. 

Получилась цельная формализованная структура:
Observation "Стадия" и жёстко связанные с ней (через атрибуты hasMember) Observation для T, N, M.

Эта конструкция и является формализованной записью подтверждений для краткой записи стадии в ресурсе Condition.

То есть, ссылка идёт не просто на Observation с тем же значением стадии, а на конструкцию из стадии и связанных с ней оценках T, N, M.

Можно было бы сразу дать ссылку на Observation с оценками T, N, M, но в MCODE решили сделать профиль для кодирования стадии с оценками T, N, M чтобы можно было хранить и передавать стадирование независимо, а не только в составе Condition'a.

**Comment by Vernir** [comment]
Возник вопрос относительно работы с параметрами, которые могут быть как отдельными патологическими состояниями, так и проявлениями другого заболевания. Пример: Гипертрофия левого желудочка, которая может быть как в рамках артериальной гипертензии (например, I11.9), так и сама по себе. В первом случае желательно предусмотреть структуру для привязки ГЛЖ к ресурсу нозологической единицы, как параметр.

**Reply by eugenekogan-spb** [reply]
Гипертрофия левого желудочка (видимо, наблюдаемая на УЗИ сердца) не является ли Observation?

**Comment by Vrodvi** [comment]
Для учёта международного опыта осуществлён поиск профилей для ресурса Condition в различных национальных руководствах по реализации FHIR.

Поиск осуществлён по официальному регистру руководств по реализации, который ведётся по адресу:
https://www.fhir.org/guides/registry/
(технически регистр ведётся как репозиторий в github по адресу https://github.com/FHIR/ig-registry - после публикации первой версии корневого российского руководства по реализации можно будет добавить его в этот регистр).

---

Все найденные профили Condition определены очень поверхностно с минимальными ограничениями и пояснениями.

Интересный вариант реализации найден в Швейцарии - там определена структура данных для ракового реестра, но все данные о диагнозе определены как несвязанные между собой Observation (а не Condition), упакованы в "разделы" документа (похоже на наследие CDA).

---

Найденные на данный момент профили ресурса Condition:

**Австралия**: http://hl7.org.au/fhir/2021Aug/StructureDefinition-au-condition.html

**США**: http://hl7.org/fhir/us/core/StructureDefinition-us-core-condition.html

**Дания**: нет профиля condition

**Швейцария**: базового профиля нет, 
есть профиль для MedNet:
https://mednet.swiss/fhir/StructureDefinition-mni-condition.html
есть профиль для ракового реестр (по структуре - документ):  
https://fhir.ch/ig/ch-crl/StructureDefinition-ch-crl-composition.html
https://fhir.ch/ig/ch-crl/StructureDefinition-ch-crl-condition-disease.html
https://fhir.ch/ig/ch-crl/StructureDefinition-ch-crl-observation-icd10.html
https://fhir.ch/ig/ch-crl/StructureDefinition-ch-crl-observation-chronicpulmonarydisease.html
https://fhir.ch/ig/ch-crl/StructureDefinition-ch-crl-observation-myocardialinfarction.html

**Нидерланды**: https://simplifier.net/nictizstu3-zib2017/zib-problem

**Германия**: https://simplifier.net/BasisprofilDE/condition-de-basis-0.2/~overview

**Италия**: нет профиля condition

**Норвегия**: нет профиля condition

**Бельгия**: https://simplifier.net/ehealthplatformfederalprofiles/beproblem

**Канада**: https://build.fhir.org/ig/HL7-Canada/ca-baseline/branches/master/StructureDefinition-profile-condition.html

**Великобритания**: нет профиля condition

**IPS** (международный проект "выписки"): https://build.fhir.org/ig/HL7/fhir-ips/StructureDefinition-Condition-uv-ips.html

---

**Reply by JustAsWalker** [reply]
**Германия (R4):** 
- https://simplifier.net/guide/MedizininformatikInitiative-ModulDiagnosen-ImplementationGuide/IGMIIKDSModulDiagnose/TechnischeImplementierung/FHIRProfile/Condition.guide.md
- https://simplifier.net/packages/de.medizininformatikinitiative.kerndatensatz.diagnose/2.0.0-alpha3/files/402364

**Comment by Vrodvi** [comment]
Профили для нозологической единицы (Condition) и случая (Encounter) переведены на R5.
Прикладываю тут в формате FSH, т.к. не понятно куда иначе их класть.
[core-R5-encounter-and-condition-resources.zip](https://github.com/fhir-ru/core/files/12371080/core-R5-encounter-and-condition-resources.zip)
@OlegPenzin @eugenekogan-spb 

Профили проверены через прогон через SUSHI последней версии. На подготовленных профилях успешно генерируется стандартный сайт с руководством по реализации - прикладываю его в качестве дополнительной демонстрации (открывать через файл index.html):
[output.zip](https://github.com/fhir-ru/core/files/12371088/output.zip)

**Reply by Vrodvi** [reply]
@OlegPenzin @eugenekogan-spb 
Новая версия профилей - исправлены все ошибки со страницы QA (quality assurence).
[core-R5-encounter-and-condition-resources.zip](https://github.com/fhir-ru/core/files/12621262/core-R5-encounter-and-condition-resources.zip)

Вот сводка со страницы QA:
![image](https://github.com/fhir-ru/core/assets/92928137/edce96e1-79ab-4985-b4d7-7074a0b8dac3)

Не понятно что там за информационное сообщение осталось и два странных замечания (warnings) про fixed[x] в расширениях - не понятно как это исправить, т.к. никаких фиксированных значений в расширениях не даётся, возможно это какая-то ошибка на их стороне:
![image](https://github.com/fhir-ru/core/assets/92928137/df814d6a-f0c0-4399-9c45-7ffdfe29d404)

**Reply by Vrodvi** [reply]
@OlegPenzin @eugenekogan-spb 

Внёс ещё небольшие правки (касаются определения псевдонимов/alias): 
[core-R5-encounter-and-condition-resources.zip](https://github.com/fhir-ru/core/files/12685222/core-R5-encounter-and-condition-resources.zip)

---

Также фиксирую рекомендации по формированию псевдонимов на основании полученного опыта:

### Рекомендуемый формат определения псевдонимов (alias) артефактов:
- наименование псевдонима должно начинаться с символа `$`
- наименование псевдонима должно быть написано в регистре UpperCamelCase (каждое слово - с заглавной буквы), разделитель между группами слов обозначать символом `_`
- наименование псевдонима должно быть написано по-русски
- наименование псевдонима должно соответствовать фасету `ИсточникАртефакта_ТипАртефакта_НаименованиеАртефакта`

#### Пример:
- определим справочник
```
CodeSystem: Nsi_Cs_1_2_643_5_1_13_13_11_1077 
Id:         nsi-cs-1-2-643-5-1-13-13-11-1077
Title:     "Виды нозологических единиц диагноза"
```
- сформируем канонический URL этого справочника (фасет канонического URL - `BaseIgUrl_ResourceType_ResourceId`)
```
http://fhir.ru/core/CodeSystem/nsi-cs-1-2-643-5-1-13-13-11-1077
```
- сформируем псевдоним для этого справочника (псевдоним устанавливаем на сформированный канонический URL)
```
Alias: $НСИ_Справочник_ВидыНозологическихЕдиницДиагноза = http://fhir.ru/core/CodeSystem/nsi-cs-1-2-643-5-1-13-13-11-1077
```
- используем сформированный псевдоним (в описании профиля и в экземпляре набора значений)
```
* diagnosis.use[rubric].coding.system = $НСИ_Справочник_ВидыНозологическихЕдиницДиагноза
```
```
* include codes from system $НСИ_Справочник_ВидыНозологическихЕдиницДиагноза
```
- убедимся, что использованный псевдоним успешно преобразован в ссылку на корректную страницу руководства по реализации
![image](https://github.com/fhir-ru/core/assets/92928137/2f88876d-c5b7-4b12-882e-f8824cf4114f)

**Reply by Vrodvi** [reply]
@OlegPenzin @eugenekogan-spb 
Последняя версия с добавлением маркдаун-файлов с описаниями для страниц профилей ресурсов encounter и condition.
[input.zip](https://github.com/fhir-ru/core/files/12835385/input.zip)

(проверить корректность формирования сайта руководства я не смог - у меня перестал работать генератор после замены рабочего ноутбука)

**Comment by eugenekogan-spb** [comment, marked as answer]
@Vrodvi вопросы о профиле CoreCondition

- для чего в профиле описан identifier, если никакие ограничения на него не наложены?
- почему в атрибуте clinicalStatus описано расширение для русских наименований, а в verificationStatus нет?
- почему набор значений клинического статуса, статуса подтверждения не описан как набор значений, а только приведен в названии?
- достаточно ли оснований для ограничения, что condition.code.coding может быть единственный?
- достаточно ли оснований для ограничения, что привязка к справочнику МКБ-10 required?

**Reply by OlegPenzin** [reply]
для чего в профиле описан identifier, если никакие ограничения на него не наложены?
- это результат добавления русскоязычного описания

почему в атрибуте clinicalStatus описано расширение для русских наименований, а в verificationStatus нет?
- русские наименования для http://hl7.org/fhir/valueset-condition-ver-status.html подготовим и добавим расширение для их указания

почему набор значений клинического статуса, статуса подтверждения не описан как набор значений, а только приведен в названии?
- проблема в алиасах, перейдем на использование прямой ссылки

достаточно ли оснований для ограничения, что condition.code.coding может быть единственный?
- первый слайс остается на код по ValueSet'у с МКБ, а остальные кодинги будут доступны на переходный период к МКБ-11 (до окончательного решения регулятора)

достаточно ли оснований для ограничения, что привязка к справочнику МКБ-10 required?
- жесткая привязка к МКБ-10 меняется на привязку к ValueSet'у с МКБ, в котором в том числе может быть и МКБ-11 (в последствие, пока только МКБ-10) + остается отдельный не решенный вопрос по версионированию профилей

**Reply by eugenekogan-spb** [reply]
Олег, спасибо за разъяснения. 
Остались такие вопросы.
1) Хотелось бы вопрос о переводе отделить от профилирования.
2) Решение о единственности МКБ (10, 11, - неважно) на уровне RuCore мне представляется неверным. Есть примеры "многоосевых" диагнозов, например в детской психиатрии. Для такого описания должны быть выставлены коды как по МКБ, так и по иным справочникам.
